src/mcp/client/streamable_http.py

```diff
@@ -83,6 +95,7 @@ def __init__(
         headers: dict[str, Any] | None = None,
         timeout: timedelta = timedelta(seconds=30),
         sse_read_timeout: timedelta = timedelta(seconds=60 * 5),
+        auth_token_provider: AuthTokenProvider | None = None,
```

Comment:
Can we add some documentation for this field to the `StreamableHTTPTransport` docstring? I.e. note whether this takes precedence over headers in `headers`, etc. Also, should we align the name with the typescript SDK (IIRC it's called OauthClientProvider there ?)